src/inverse/SIA2D/VJPs.jl

```diff
@@ -67,3 +68,41 @@ function VJP_λ_∂surface_V∂θ(VJPMode::DiscreteVJP, λx, λy, H, θ, simulat
     λ_∂V∂H = VJP_λ_∂surface_V∂θ_discrete(λx, λy, H, θ, simulation, t)
     return λ_∂V∂H, nothing
 end
+
+
+function MB_wrapper!(MB, H, simulation, glacier)
+    model = simulation.model
+    cache = simulation.cache
+    glacier.S .= glacier.B .+ H
+
+    # Below we call the functions inside MB_timestep! manually
+    # This is because get_cumulative_climate! cannot be differentiated with Enzyme, so it is called beforehand in the VJP function to retrieve the cumulative climate
+    downscale_2D_climate!(glacier)
+    cache.iceflow.MB .= compute_MB(model.mass_balance, glacier.climate.climate_2D_step)
+
+    apply_MB_mask!(H, cache.iceflow)
+    MB .= simulation.cache.iceflow.MB
+end
+function VJP_λ_∂MB∂H(VJPMode::EnzymeVJP, λ, H, simulation::Simulation, glacier, t)
+    # Differentiation of get_cumulative_climate! with Enzyme yields an error
+    # Since it isn't involved in the gradient computation (doesn't depend on H), it can be computed beforehand
+    get_cumulative_climate!(glacier.climate, t, simulation.parameters.solver.step)
+
+    _simulation = Enzyme.make_zero(simulation)
+    _glacier = Enzyme.make_zero(glacier)
+    λ_∂MB∂H = Enzyme.make_zero(H)
+    MB = Enzyme.make_zero(H)
+    λH = deepcopy(λ) # Need to copy because Enzyme changes the backward gradient in-place
+    Enzyme.autodiff(
+        Reverse, MB_wrapper!, Const,
+        Duplicated(MB, λH),
+        Duplicated(H, λ_∂MB∂H),
+        Duplicated(simulation, _simulation),
+        Duplicated(glacier, _glacier),
+    )
+    return λ + λ_∂MB∂H
```

Comment:
I find a bit confusing that this returns the adjoint itself and the contribution of MB at the same time. I left another comment in `gradient.jl` regarding this point, but wouldn't make more sente to return just `λ_∂MB∂H` and then in the gradient calculation have a `.+=` instead of a `.=`?